### DISCUSSION TITLE
Problem running unittest

### DISCUSSION THREAD

**Discussion by ellabaron** [1200wd/bitcoinlib#376] [Q&A] [answered]
Hi,

I've followed your instruction for installing from source on Linux. Everything seems to be fine till I try to execute unittest command. Here is what I see:

>python -m unittest
Could not import all modules. Error: No module named 'mysql'
UNITTESTS_FULL_DATABASE_TEST: False
E
======================================================================
ERROR: tests (unittest.loader._FailedTest.tests)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 440, in _find_test_path
    package = self._get_module_from_name(name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/home/ebaron/tools/bitcoinlib/tests/__init__.py", line 24, in <module>
    import tests.test_wallets
  File "/home/ebaron/tools/bitcoinlib/tests/test_wallets.py", line 137, in <module>
    @parameterized_class(*params)
     ^^^^^^^^^^^^^^^^^^^
NameError: name 'parameterized_class' is not defined

----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)

Can you please tell me what might be a problem here?

**Comment by mccwdev** [comment, marked as answer]
For development you have to install the mysql and postgresql libraries.

```python -m pip install .[dev]```

See https://bitcoinlib.readthedocs.io/en/latest/source/_static/manuals.install.html

**Reply by ellabaron** [reply]
Thank you. This was exactly my problem. I misunderstood "[dev]". I thought it was something optional and ran install without it. Anyway, it is able to run the unit tests now.
